### PR TITLE
FCE-552 Add constraints to getUserMedia

### PR DESCRIPTION
**Pull request by kamil-stasiak**
#### This configuration remains optional.
`<FishjamProvider>`  
Default constraints will be used if no configuration is provided.

#### Users can override default values.  
`<FishjamProvider deviceManagerConfig={{ videoTrackConstraints: { frameRate: 30 } }}>`  
User configurations will be merged with the default constraints.

#### Users can disable audio and/or video.  
`<FishjamProvider deviceManagerConfig={{ videoTrackConstraints: false, audioTrackConstraints: false }}>`

## Motivation and Context

To set consistent constraints and provide an option to override them.

## Types of Changes

Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: packages/react-client/src/fishjamProvider.tsx [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
 import { FishjamContext } from "./hooks/useFishjamContext";
 import { DeviceManager } from "./DeviceManager";
 import { useParticipantStatus } from "./hooks/useParticipantStatus";
 import { useFishjamClientState } from "./hooks/useFishjamClientState";
+import { AUDIO_TRACK_CONSTRAINTS, VIDEO_TRACK_CONSTRAINTS } from "./constraints";
 
 interface FishjamProviderProps extends PropsWithChildren {
   config?: { reconnect?: ReconnectConfig | boolean };
-  deviceManagerDefaultConfig?: DeviceManagerConfig;
+  deviceManagerConfig?: DeviceManagerUserConfig;
```
Comment (czerwiukk) [inline packages/react-client/src/fishjamProvider.tsx:14]: **suggestion:** user doesn't know anything about a device manager. the property could be just named `defaultConstraints` or `constraints`

File: packages/react-client/src/constraints.ts [hunk 2]
```diff
@@ -28,28 +28,37 @@ export const SCREEN_SHARING_MEDIA_CONSTRAINTS: MediaStreamConstraints = {
...
-  }
-  return constraint;
+  defaultConstraints: MediaTrackConstraints,
+  userConstraints?: boolean | MediaTrackConstraints,
+): boolean | MediaTrackConstraints => {
+  if (userConstraints === false) return false;
+
+  if (!userConstraints) return defaultConstraints;
+
+  return userConstraints;
```
Comment (czerwiukk) [inline packages/react-client/src/constraints.ts:38]: **nitpick:** 
the whole function could be replaced with this expression
```
userConstraints ?? defaultConstraints
```

File: packages/react-client/src/constraints.ts [hunk 3]
```diff
@@ -28,28 +28,37 @@ export const SCREEN_SHARING_MEDIA_CONSTRAINTS: MediaStreamConstraints = {
...
 export const prepareConstraints = (
   deviceIdToStart: string | undefined,
-  constraints: MediaTrackConstraints | undefined,
-): MediaTrackConstraints | undefined => {
+  constraints: MediaTrackConstraints | undefined | boolean,
+): MediaTrackConstraints | undefined | boolean => {
   if (!deviceIdToStart) return constraints;
 
-  return { ...constraints, deviceId: { ideal: deviceIdToStart } };
+  // does not start this device
```
Comment (czerwiukk) [inline packages/react-client/src/constraints.ts:58]: **question:** what does it mean?

File: packages/react-client/src/fishjamProvider.tsx (hunk 1 above)
Comment (czerwiukk) [reply packages/react-client/src/fishjamProvider.tsx:14]: storage, if actually needed, could be an another property such as persistence options or sth

File: packages/react-client/src/constraints.ts (hunk 3 above)
Comment (kamil-stasiak) [reply packages/react-client/src/constraints.ts]: I rephrased it to:
```
// The resulting stream will not contain a track of this type,
// which means that the device will not be activated.
```

File: packages/react-client/src/fishjamProvider.tsx (hunk 1 above)
Comment (kamil-stasiak) [reply packages/react-client/src/fishjamProvider.tsx]: WDYT about it?

```
interface FishjamProviderProps extends PropsWithChildren {
  config?: { reconnect?: ReconnectConfig | boolean };
  constraints?: Pick<MediaStreamConstraints, "audio" | "video">;
  persistLastDevice?: boolean | PersistLastDeviceHandlers;
}
```

Comment (czerwiukk) [reply packages/react-client/src/fishjamProvider.tsx]: please unpack reconnect from config and i'm satisfied 💃

Comment (kamil-stasiak) [reply packages/react-client/src/fishjamProvider.tsx]: updated ✅

**Review by czerwiukk — APPROVED** [review]

**Pull request merged**